Challenge2/src/agent.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Finding,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+  TransactionEvent,
+  getEthersProvider,
+  ethers,
+} from "forta-agent";
+import { Provider } from "@ethersproject/providers";
+import { LRUCache } from "lru-cache";
+import { UNI_SWAP_EVENT_ABI, UNI_POOL_FUNCTIONS_ABI, UNI_FACTORY_ADDRESS, UNI_INIT_CODE_HASH } from "./constants";
+
+const addressIsUniCache = new LRUCache<string, boolean>({ max: 100000 });
+
+const getPoolValues = async (
+  provider: Provider,
+  uniPoolFunctionsAbi: string[],
+  poolAddress: string,
+  txEvent: TransactionEvent
+) => {
+  const poolContract = new ethers.Contract(poolAddress, uniPoolFunctionsAbi, provider);
+  const poolValues: any[] = await Promise.all([
+    poolContract.token0({ blockTag: txEvent.blockNumber }),
+    poolContract.token1({ blockTag: txEvent.blockNumber }),
+    poolContract.fee({ blockTag: txEvent.blockNumber }),
+  ]);
+  return poolValues;
+};
+
+export const getRealPoolAddress = (uniFactoryAddress: string, uniInitCode: string, interceptedPoolValues: any[]) => {
```

Comment:
Same comment regarding usage of `any` for `interceptedPoolValues: any[]`.

Reply:
Also, this function, along with `getPoolValues` can be in their own file and imported here. Just to clean up this file a bit. A `utils.ts` file.

Another note: it is common for us to have (and we often times we suggest) interns implement this type of functionality as its own class. That is because they "tied" to one another, as one function actually depends on the output of the other. Then afterwards, this allows you to create a test suite for this class in isolation and then mock that class in the test suite for the bot. If you're curious about what I'm referring to, feel to check out some other interns submissions for challenge 02.